sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/Signer.cs

```diff
@@ -106,7 +106,19 @@ public static void SignRequest(IRequestContext requestContext)
 
             // credentials would be null in the case of anonymous users getting public resources from S3
             if (immutableCredentials == null && requestContext.Signer.RequiresCredentials)
+            {
                 return;
+            }
+
+            // The signer interface expects immutable credentials, which does not match the AWSToken structure.
+            // We'll manually set a property so that the bearer token still works as expected.
+            if (requestContext.Signer is BearerTokenSigner tokenSigner)
+            {
+                if (requestContext.Identity is AWSToken resolvedToken)
+                {
+                    tokenSigner.ResolvedToken = resolvedToken.Token;
```

Comment:
We are mutating state on the bearer token signer which is being use across threads. We need a safer way to do this.

Reply:
I updated the signer to use a lock here: https://github.com/aws/aws-sdk-net/pull/3562/commits/8f86f7c0a8b212e51a0dfd29fe586b6fa8d34ae6#diff-4f67ae8bdcfede242a71fccc1853340fa9affa50ebf705fdff16b52438253293

Let me know if there's a better way to do this.

Reply:
See my comment above about switching the `Sign` method to take an an `BaseIdentity` for `credentials`.